bokehjs/src/lib/models/widgets/date_picker.ts

```diff
@@ -1,51 +1,17 @@
-import Pikaday from "pikaday"
+import flatpickr from "flatpickr"
 
 import {InputWidget, InputWidgetView} from "./input_widget"
 import {input} from "core/dom"
 import * as p from "core/properties"
 import {bk_input} from "styles/widgets/inputs"
-import "styles/widgets/pikaday"
-
-Pikaday.prototype.adjustPosition = function(this: Pikaday & {_o: Pikaday.PikadayOptions}): void {
-  if (this._o.container)
-    return
-
-  this.el.style.position = 'absolute'
-
-  const field = this._o.trigger!
-  const width = this.el.offsetWidth
-  const height = this.el.offsetHeight
-  const viewportWidth = window.innerWidth || document.documentElement!.clientWidth
-  const viewportHeight = window.innerHeight || document.documentElement!.clientHeight
-  const scrollTop = window.pageYOffset || document.body.scrollTop || document.documentElement!.scrollTop
-
-  const clientRect = field.getBoundingClientRect()
-  let left = clientRect.left + window.pageXOffset
-  let top = clientRect.bottom + window.pageYOffset
-
-  // adjust left/top origin to .bk-root
-  left -= this.el.parentElement!.offsetLeft
-  top -= this.el.parentElement!.offsetTop
-
-  // default position is bottom & left
-  if ((this._o.reposition && left + width > viewportWidth) ||
-      (this._o.position!.indexOf('right') > -1 && left - width + field.offsetWidth > 0))
-    left = left - width + field.offsetWidth
-
-  if ((this._o.reposition && top + height > viewportHeight + scrollTop) ||
-      (this._o.position!.indexOf('top') > -1 && top - height - field.offsetHeight > 0))
-    top = top - height - field.offsetHeight
-
-  this.el.style.left = left + 'px'
-  this.el.style.top = top + 'px'
-}
+import "styles/widgets/flatpickr"
 
 export class DatePickerView extends InputWidgetView {
   model: DatePicker
 
   protected input_el: HTMLInputElement
 
-  private _picker: Pikaday
+  private _picker: any
```

Comment:
`flatpickr` has type declarations, so this and other `any`s can be replaced with more precise types.